e2e_tests/azure_function/test_azure_function_deployment.py

```diff
@@ -0,0 +1,74 @@
+import logging
+import subprocess
+import time
+import uuid
+
+import requests
+
+from e2e_tests.cli_operations import delete_deployment
+
+logger = logging.getLogger('bentoml.test')
+
+
+def test_azure_function_deployment(iris_clf_service):
+    random_hash = uuid.uuid4().hex[:6]
+    deployment_name = f'test-azure-{random_hash}'
+    create_deployment_command = [
+        'bentoml',
+        'azure-function',
+        'deploy',
+        deployment_name,
+        '-b',
+        iris_clf_service,
+        '--location',
+        'westus',
+        '--max-burst',
+        '2',
+        '--function-auth-level',
+        'anonymous',
+        '--debug',
+    ]
+    try:
+        logger.info(f'Deploying {deployment_name} to Azure function')
+        with subprocess.Popen(
+            create_deployment_command, stdout=subprocess.PIPE, stderr=subprocess.PIPE
+        ) as proc:
+            stdout = proc.stdout.read().decode('utf-8')
+        logger.info(stdout)
+        if 'Failed to create Azure function deployment' in stdout:
+            raise Exception('Failed to create Azure function deployment')
```

Comment:
assert  'Failed to create Azure function deployment' not in stdout, "Azure function deployment creation should success"